otter/cloud_client/__init__.py

```diff
@@ -705,21 +708,45 @@ def get_clbs():
         success=lambda (response, body): body['loadBalancers'])
 
 
-def get_clb_node_feed(lb_id, node_id):
-    """Get the atom feed associated with a CLB node. Returns feed as str."""
+def _node_feed_request(lb_id, node_id, params):
     return service_request(
-        ServiceType.CLOUD_LOAD_BALANCERS,
-        'GET',
+        ServiceType.CLOUD_LOAD_BALANCERS, 'GET',
         append_segments('loadbalancers', str(lb_id), 'nodes',
                         '{}.atom'.format(node_id)),
+        params=params,
         json_response=False
     ).on(
         error=_only_json_api_errors(
             lambda c, b: _process_clb_api_error(c, b, lb_id))
     ).on(
         log_success_response('request-get-clb-node-feed', identity)
-    ).on(
-        success=lambda (response, body): body)
+    ).on(itemgetter(1))
+
+
+@do
+def get_clb_node_feed(lb_id, node_id):
+    """
+    Get the atom feed associated with a CLB node.
+
+    :param int lb_id: Cloud Load balancer ID
+    :param int node_id: Node ID of in loadbalancer node
+
+    :returns: ``list`` of atom entry :class:`Element`
+    :rtype: ``list``
+    """
+    all_entries = []
+    params = {}
+    while True:
+        feed_str = yield _node_feed_request(lb_id, node_id, params)
+        feed = atom.parse(feed_str)
+        entries = atom.entries(feed)
+        if entries == []:
+            yield do_return(all_entries)
+        all_entries.extend(entries)
+        next_link = atom.next_link(feed)
+        if not next_link:
+            yield do_return(all_entries)
+        params = parse_qs(urlparse(next_link).query)
```

Comment:
Technically speaking, this is subtly incorrect.  In a REST protocol like Atom, the "next link" may be any valid HTTPS URI; you shouldn't be parsing it or trying to understand it here, just issuing a `GET` to the exact URL that you found in the feed.  For example, it's valid for a feed to say `/foo/bar?page=2` or `/foo/bar/page/2` or even `https://example.com/e8e0e8ce795e3bee11f0`.  Nothing says that it's just the query string that changes.

I think this is fine for now given the way that cloud feeds' API works and is documented, but would you mind at least documenting the technical debt here with an issue?